llvm/test/CodeGen/SBF/sub_reversed_immidiate.ll

```diff
@@ -0,0 +1,82 @@
+; RUN: llc < %s -march=sbf -mattr=+alu32 | FileCheck %s
+;
+; Source:
+; void test_func_64(long * vec) {
+;     vec[0] = 50 - vec[0];
+;     vec[1] = vec[3] - vec[2];
+; }
+;
+; void test_func_32(int * vec) {
+;     vec[0] = 50 - vec[0];
+;     vec[1] = vec[3] - vec[2];
+; }
+;
+; Compilation flag:
+; clang -S -emit-llvm test.c
+
+
+; Function Attrs: noinline nounwind optnone ssp uwtable(sync)
+define void @test_func_64(ptr noundef %vec) #0 {
```

Comment:
All of the new test cases are too verbose/complicated and should simplified. We only need to unit test a single instruction, so the tests should be very narrowly written that way. We want minimal testcases-- a single LLVM IR instruction is all that is needed to cover the current functionality. For example, a test for `sub64` can be succintly written as follows:
```
; Function Attrs: norecurse nounwind readnone
define dso_local i64 @test_sub64_ri(i64 %a) local_unnamed_addr #0 {
entry:
  %t = sub nsw i64 50, %a
; CHECK: sub64 r{{[0-9]+}}, 50
  ret i64 %t
}
```
Note also that I use FileCheck's regex matching on the register name to make it slightly less impervious to register allocation changes over time.

Similarly, write the other test cases as shown above.

Also, you can just fold the negate tests into `sub_reversed_immidiate.ll` and remove `neg_inst_with_sub.ll`.
Finally, please rename `sub_reversed_immidiate.ll` -> `sub_reversed_immediate.ll`.